Facade_Engine/Query/GeneralMaterialTakeoff.cs

```diff
@@ -0,0 +1,189 @@
+            }
+            takeoffs.Add(Physical.Query.IGeneralMaterialTakeoff(opening.OpeningConstruction, opening.ComponentAreas().Item1));
+
+            return Matter.Compute.AggregateGeneralMaterialTakeoff(takeoffs);
+        }
+
+        /***************************************************/
+
+        [Description("Gets the general material takeoff from the FrameEdge.")]
+        [Input("edge", "The frame edge to extract the general material takeoff from.")]
+        [Output("genTakeoff", "The general material takeoff for the FrameEdge.")]
+        public static GeneralMaterialTakeoff GeneralMaterialTakeoff(this FrameEdge edge)
+        {
+            if (edge == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"Cannot query the {nameof(GeneralMaterialTakeoff)} of a null {nameof(FrameEdge)}.");
+                return null;
+            }
+
+            if (edge.Curve == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"Cannot query the {nameof(GeneralMaterialTakeoff)} of a {nameof(FrameEdge)} with null geometry.");
+                return null;
+            }
+
+            if (edge.FrameEdgeProperty == null)
+            {
+                Engine.Base.Compute.RecordError($"The {nameof(GeneralMaterialTakeoff)} could not be queried as no {nameof(FrameEdgeProperty)} has been assigned to the {nameof(FrameEdge)}.");
+                return null;
+            }
+
+            VolumetricMaterialTakeoff volTakeoff = Matter.Create.VolumetricMaterialTakeoff(edge.MaterialComposition(), edge.SolidVolume());
+            
+            return Matter.Create.GeneralMaterialTakeoff(volTakeoff);
+        }
+
+
+        /***************************************************/
+        /**** Public Methods - Interface                ****/
+        /***************************************************/
+
+        [Description("Gets the volumetric material takeoff from the IOpening.")]
+        [Input("opening", "The physical opening object to extract the general material takeoff from.")]
+        [Output("genTakeoff", "The general material takeoff for the opening, made of up the volume and materiality of the construction and surface area.")]
+        public static GeneralMaterialTakeoff IGeneralMaterialTakeoff(this IOpening opening)
+        {
+            return GeneralMaterialTakeoff(opening as dynamic);
+        }
+
+
+        /***************************************************/
+        /**** Private Methods - Fallbacks               ****/
+        /***************************************************/
+
+        [Description("Gets the volumetric material takeoff from the object.")]
+        [Input("obj", "The object to extract the material takeoff from.")]
+        [Output("genTakeoff", "The volumetric material takeoff for the opening, made of up the volume and materiality of the construction and surface area.")]
+        public static GeneralMaterialTakeoff GeneralMaterialTakeoff(this object obj)
```

Comment:
This should be a `private` method per the comment and [convention](https://github.com/BHoM/BHoM_Engine/blob/develop/Geometry_Engine/Query/EndPoint.cs#L119).
```suggestion
        private static GeneralMaterialTakeoff GeneralMaterialTakeoff(this object obj)
```

Reply:
Change made (oops)